# DISCOVERING  USED CAR BRANDS IN EBAY WEBSITE

In this project, we have car data set which is obtained from eBay website. There is some information about used cars in this data set. Our aim in this project is analyzing these used cars' brands and prices.

### Opening Data

In [1]:
import pandas as pd
import numpy as np

In [2]:
autos= pd.read_csv("autos.csv",encoding= 'Latin-1')

In [3]:
#We explore the data set
autos.info()
autos.head(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2  2016-03-26 18:57:24                  Volkswagen_Golf_1.6_United  privat   

  offerType   price   abtest vehicleType  yearOfRegistration    gearbox  \
0   Angebot  $5,000  control         bus                2004    manuell   
1   Angebot  $8,500  control   limousine                1997  automatik   
2   Angebot  $8,990     test   limousine                2009    manuell   

   powerPS   model   odometer  monthOfRegistration fuelType       brand  \
0      158  andere  150,000km                    3      lpg     peugeot   
1      286     7er  150,000km                    6   benzin         bmw   
2      102    golf   70,000km                    7   benzin  volkswagen   

  notRepairedDamage          dateCreated  nrOfPictures  postalCode  \
0              nein  2016-03-26 00:00:00             0       79588   
1              nein  2016-04-04 00:00:00             0       71034   
2              nein  2016-03-26 00:00:00             0       35394   

              lastSeen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37

There are some missing values. There are some objects which can obstract some analysis. This things must be cleaned to analyze data set perfectly. 

In [4]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

You can see the column names below. In this step, we will change the column names from the camelcase to snakecase.

In [5]:
autos.rename(columns={"yearOfRegistration":"registration_year"}).head(1)

dateCrawled                              name  seller offerType  \
0  2016-03-26 17:47:46  Peugeot_807_160_NAVTECH_ON_BOARD  privat   Angebot   

    price   abtest vehicleType  registration_year  gearbox  powerPS   model  \
0  $5,000  control         bus               2004  manuell      158  andere   

    odometer  monthOfRegistration fuelType    brand notRepairedDamage  \
0  150,000km                    3      lpg  peugeot              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-26 00:00:00             0       79588  2016-04-06 06:45:54

In [6]:
autos.columns= ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen']

In [7]:
#This is our new version of the data set.
autos.head(2)

date_crawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   

  offer_type   price  ab_test vehicle_type  registration_year    gearbox  \
0    Angebot  $5,000  control          bus               2004    manuell   
1    Angebot  $8,500  control    limousine               1997  automatik   

   power_ps   model   odometer  registration_month fuel_type    brand  \
0       158  andere  150,000km                   3       lpg  peugeot   
1       286     7er  150,000km                   6    benzin      bmw   

  unrepaired_damage           ad_created  num_photos  postal_code  \
0              nein  2016-03-26 00:00:00           0        79588   
1              nein  2016-04-04 00:00:00           0        71034   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08

### Exploring and Cleaning Data

Let's start discover the data more.

In [8]:
autos.describe(include='all')


date_crawled         name  seller offer_type  price ab_test  \
count                 50000        50000   50000      50000  50000   50000   
unique                48213        38754       2          2   2357       2   
top     2016-03-21 16:37:21  Ford_Fiesta  privat    Angebot     $0    test   
freq                      3           78   49999      49999   1421   25756   
mean                    NaN          NaN     NaN        NaN    NaN     NaN   
std                     NaN          NaN     NaN        NaN    NaN     NaN   
min                     NaN          NaN     NaN        NaN    NaN     NaN   
25%                     NaN          NaN     NaN        NaN    NaN     NaN   
50%                     NaN          NaN     NaN        NaN    NaN     NaN   
75%                     NaN          NaN     NaN        NaN    NaN     NaN   
max                     NaN          NaN     NaN        NaN    NaN     NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  num_photos   postal_code            last_seen  
count                 50000     50000.0  50000.000000                50000  
unique                   76         NaN           NaN                39481  
top     2016-04-03 00:00:00         NaN           NaN  2016-04-07 06:17:27  
freq                   1946         NaN           NaN                    8  
mean                    NaN         0.0  50813.627300                  NaN  
std                     NaN         0.0  25779.747957                  NaN  
min                     NaN         0.0   1067.000000                  NaN  
25%                     NaN         0.0  30451.000000                  NaN  
50%                     NaN         0.0  49577.000000                  NaN  
75%                     NaN         0.0  71540.000000                  NaN  
max                     NaN         0.0  99998.000000                  NaN

Our initial observations:
<br>
There are a number of text columns where all (or nearly all) of the values are the same:
<br>
*seller
<br>
*offer_type
<br>
The 'num_photos' column looks odd, we'll need to investigate this further.

In [9]:
autos['num_photos'].value_counts()

0    50000
Name: num_photos, dtype: int64

The 'num_photos' column has only zero value, so it useless for our analyze. Right now, we can drop this column. Also, the 'seller' and 'offer_type' columns have string values and they are far away to be useful for our aim. We can drop all off these colums.

In [10]:
autos= autos.drop(['num_photos','seller', 'offer_type'], axis=1)

After this step, we might prepare some columns to analyze. The 'price' and 'odometer' colums have different characters and are stored as string. However, we need integer values to analyze. 

In [11]:
autos["price"]=autos["price"].str.replace("$","").str.replace(",","").astype(float)

In [12]:
autos["price"].head()

0    5000.0
1    8500.0
2    8990.0
3    4350.0
4    1350.0
Name: price, dtype: float64

In [13]:
autos["odometer"]= autos["odometer"].str.replace("km","").str.replace(",","").astype(int)

In [14]:
autos["odometer"].head()

0    150000
1    150000
2     70000
3     70000
4    150000
Name: odometer, dtype: int32

In [15]:
autos.rename({"odometer": "odometer_km"},axis=1, inplace= True)# We changed the column name to decribe the unit of distance

In [16]:
autos.describe(include='all')

date_crawled         name         price ab_test vehicle_type  \
count                 50000        50000  5.000000e+04   50000        44905   
unique                48213        38754           NaN       2            8   
top     2016-03-21 16:37:21  Ford_Fiesta           NaN    test    limousine   
freq                      3           78           NaN   25756        12859   
mean                    NaN          NaN  9.840044e+03     NaN          NaN   
std                     NaN          NaN  4.811044e+05     NaN          NaN   
min                     NaN          NaN  0.000000e+00     NaN          NaN   
25%                     NaN          NaN  1.100000e+03     NaN          NaN   
50%                     NaN          NaN  2.950000e+03     NaN          NaN   
75%                     NaN          NaN  7.200000e+03     NaN          NaN   
max                     NaN          NaN  1.000000e+08     NaN          NaN   

        registration_year  gearbox      power_ps  model    odometer_km  \
count        50000.000000    47320  50000.000000  47242   50000.000000   
unique                NaN        2           NaN    245            NaN   
top                   NaN  manuell           NaN   golf            NaN   
freq                  NaN    36993           NaN   4024            NaN   
mean          2005.073280      NaN    116.355920    NaN  125732.700000   
std            105.712813      NaN    209.216627    NaN   40042.211706   
min           1000.000000      NaN      0.000000    NaN    5000.000000   
25%           1999.000000      NaN     70.000000    NaN  125000.000000   
50%           2003.000000      NaN    105.000000    NaN  150000.000000   
75%           2008.000000      NaN    150.000000    NaN  150000.000000   
max           9999.000000      NaN  17700.000000    NaN  150000.000000   

        registration_month fuel_type       brand unrepaired_damage  \
count         50000.000000     45518       50000             40171   
unique                 NaN         7          40                 2   
top                    NaN    benzin  volkswagen              nein   
freq                   NaN     30107       10687             35232   
mean              5.723360       NaN         NaN               NaN   
std               3.711984       NaN         NaN               NaN   
min               0.000000       NaN         NaN               NaN   
25%               3.000000       NaN         NaN               NaN   
50%               6.000000       NaN         NaN               NaN   
75%               9.000000       NaN         NaN               NaN   
max              12.000000       NaN         NaN               NaN   

                 ad_created   postal_code            last_seen  
count                 50000  50000.000000                50000  
unique                   76           NaN                39481  
top     2016-04-03 00:00:00           NaN  2016-04-07 06:17:27  
freq                   1946           NaN                    8  
mean                    NaN  50813.627300                  NaN  
std                     NaN  25779.747957                  NaN  
min                     NaN   1067.000000                  NaN  
25%                     NaN  30451.000000                  NaN  
50%                     NaN  49577.000000                  NaN  
75%                     NaN  71540.000000                  NaN  
max                     NaN  99998.000000                  NaN

### Exploring 'price' and 'odometer_km' Columns

In [17]:
autos["price"].unique().shape

(2357,)

In [18]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [19]:
price=autos["price"].value_counts()

In [20]:
price.head()

0.0       1421
500.0      781
1500.0     734
2500.0     643
1200.0     639
Name: price, dtype: int64

In [21]:
price.head(20).sort_index(ascending=True)

0.0       1421
300.0      384
500.0      781
600.0      531
650.0      419
700.0      395
750.0      433
800.0      498
850.0      410
900.0      420
950.0      379
999.0      434
1000.0     639
1200.0     639
1500.0     734
2000.0     460
2200.0     382
2500.0     643
3500.0     498
4500.0     394
Name: price, dtype: int64

In 'price' column, there are inconsistent values. There is not possibility for prices to be more than 350.000 dollars for used cars. If we clean this data, we can get more sensible results.

In [22]:
autos= autos[autos["price"].between(1,351000)]# We have cleaned the odd values from the 'price' column.

In [23]:
autos["price"].describe()

count     48565.000000
mean       5888.935591
std        9059.854754
min           1.000000
25%        1200.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price, dtype: float64

In [24]:
autos["odometer_km"].unique().shape

(13,)

In [25]:
autos["odometer_km"].describe()


count     48565.000000
mean     125770.101925
std       39788.636804
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [26]:
autos["odometer_km"].value_counts()

150000    31414
125000     5057
100000     2115
90000      1734
80000      1415
70000      1217
60000      1155
50000      1012
5000        836
40000       815
30000       780
20000       762
10000       253
Name: odometer_km, dtype: int64

When we observe the price data, we see that there are some extra ordinary values. We removed this values. 
In odometer data there are not extraordinary values.

### Exploring the Date Columns

There are 5 different columns which show different dates about ads.
<br>
date_crawled
<br>
ad_created
<br>
last_seen
<br>
registration_year
<br>
registration_mounth
<br>
Let's explore these columns.

In [27]:
autos[['date_crawled','ad_created','last_seen','registration_year','registration_month']][0:5]

date_crawled           ad_created            last_seen  \
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54   
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08   
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37   
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28   
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50   

   registration_year  registration_month  
0               2004                   3  
1               1997                   6  
2               2009                   7  
3               2007                   6  
4               2003                   7

The 'date_crawled' column shows: When this ad was crawled first.
<br>
The 'ad_created' column shows: When this ad was created on eBay listing.
<br>
The 'last_seen' column shows: When the crawler saw this ad on website.
<br>
The 'registration_year' column shows: Which year the car was registered.
<br>
The 'registration_month' column shows: Which month the car was registered.

In [28]:
(autos['date_crawled'].str[:10]
.value_counts(normalize=True, dropna=False)
.sort_values())

2016-04-07    0.001400
2016-04-06    0.003171
2016-03-18    0.012911
2016-04-05    0.013096
2016-03-06    0.014043
2016-03-13    0.015670
2016-03-05    0.025327
2016-03-24    0.029342
2016-03-16    0.029610
2016-03-27    0.031092
2016-03-25    0.031607
2016-03-17    0.031628
2016-03-31    0.031834
2016-03-10    0.032184
2016-03-26    0.032204
2016-03-23    0.032225
2016-03-11    0.032575
2016-03-22    0.032987
2016-03-09    0.033090
2016-03-08    0.033296
2016-03-30    0.033687
2016-04-01    0.033687
2016-03-29    0.034099
2016-03-15    0.034284
2016-03-19    0.034778
2016-03-28    0.034860
2016-04-02    0.035478
2016-03-07    0.036014
2016-04-04    0.036487
2016-03-14    0.036549
2016-03-12    0.036920
2016-03-21    0.037373
2016-03-20    0.037887
2016-04-03    0.038608
Name: date_crawled, dtype: float64

In [29]:
(autos["date_crawled"].str[:10]
 .value_counts(normalize=True, dropna= False)
 .sort_index(ascending=True))

2016-03-05    0.025327
2016-03-06    0.014043
2016-03-07    0.036014
2016-03-08    0.033296
2016-03-09    0.033090
2016-03-10    0.032184
2016-03-11    0.032575
2016-03-12    0.036920
2016-03-13    0.015670
2016-03-14    0.036549
2016-03-15    0.034284
2016-03-16    0.029610
2016-03-17    0.031628
2016-03-18    0.012911
2016-03-19    0.034778
2016-03-20    0.037887
2016-03-21    0.037373
2016-03-22    0.032987
2016-03-23    0.032225
2016-03-24    0.029342
2016-03-25    0.031607
2016-03-26    0.032204
2016-03-27    0.031092
2016-03-28    0.034860
2016-03-29    0.034099
2016-03-30    0.033687
2016-03-31    0.031834
2016-04-01    0.033687
2016-04-02    0.035478
2016-04-03    0.038608
2016-04-04    0.036487
2016-04-05    0.013096
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64

In [30]:
(autos["ad_created"].str[:10]
 .value_counts(normalize=True, dropna=False)
 .sort_index(ascending=True)).head()

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
Name: ad_created, dtype: float64

In [31]:
(autos["ad_created"].str[:10]
 .value_counts(normalize=True, dropna=False)
 .sort_index(ascending=True)).tail()

2016-04-03    0.038855
2016-04-04    0.036858
2016-04-05    0.011819
2016-04-06    0.003253
2016-04-07    0.001256
Name: ad_created, dtype: float64

In [32]:
(autos["last_seen"].str[:10]
 .value_counts(normalize= True, dropna=False)
 .sort_index(ascending=True))

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64

In [33]:
autos["registration_year"].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

In [34]:
autos['registration_year'].value_counts().head()

2000    3156
2005    2936
1999    2897
2004    2703
2003    2699
Name: registration_year, dtype: int64

In [35]:
autos['registration_year'].value_counts().tail()

9000    1
1948    1
1000    1
1931    1
1952    1
Name: registration_year, dtype: int64

The year that the car was first registered will likely indicate the age of the car. Looking at this column, we note some odd values. The minimum value is 1000, long before cars were invented and the maximum is 9999, many years into the future.

In [36]:
autos['registration_month'].value_counts().sort_index()

0     4480
1     3219
2     2937
3     5003
4     4036
5     4031
6     4271
7     3857
8     3126
9     3330
10    3588
11    3313
12    3374
Name: registration_month, dtype: int64

We can see the zero values for 'registration_month' column, but we want to focus on the year of registration to analyze. Let's clean the odd values for the 'registration_year' column.

In [37]:
autos = autos[autos["registration_year"].between(1900,2016)]

In [38]:
autos["registration_year"].value_counts(normalize=True).head(10)

2000    0.067608
2005    0.062895
1999    0.062060
2004    0.057904
2003    0.057818
2006    0.057197
2001    0.056468
2002    0.053255
1998    0.050620
2007    0.048778
Name: registration_year, dtype: float64

### Exploring the Brands

In [39]:
autos["brand"].value_counts()

volkswagen        9862
bmw               5137
opel              5022
mercedes_benz     4503
audi              4041
ford              3263
renault           2201
peugeot           1393
fiat              1197
seat               853
skoda              766
nissan             713
mazda              709
smart              661
citroen            654
toyota             593
hyundai            468
sonstige_autos     458
volvo              427
mini               409
mitsubishi         384
honda              366
kia                330
alfa_romeo         310
porsche            286
suzuki             277
chevrolet          266
chrysler           164
dacia              123
daihatsu           117
jeep               106
subaru             100
land_rover          98
saab                77
jaguar              73
daewoo              70
trabant             65
rover               62
lancia              50
lada                27
Name: brand, dtype: int64

We see that there is a lot of brand. We can prefer the first ten brand and last ten brand to analyze.

In [40]:
first_ten= autos["brand"].value_counts()

In [41]:
first_ten= first_ten[:10].index

In [42]:
first_ten

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat'],
      dtype='object')

In [43]:
first_ten_brand = {}

for brand in first_ten:
    brand_only= autos[autos["brand"]== brand ]
    mean_price= brand_only["price"].mean()
    first_ten_brand[brand]=int(mean_price)
   
    

In [44]:
first_ten_brand

{'volkswagen': 5402,
 'bmw': 8332,
 'opel': 2975,
 'mercedes_benz': 8628,
 'audi': 9336,
 'ford': 3749,
 'renault': 2474,
 'peugeot': 3094,
 'fiat': 2813,
 'seat': 4397}

In [45]:
last_ten= autos["brand"].value_counts()
last_ten= last_ten[-10:].index
last_ten

Index(['jeep', 'subaru', 'land_rover', 'saab', 'jaguar', 'daewoo', 'trabant',
       'rover', 'lancia', 'lada'],
      dtype='object')

In [46]:
last_ten_brand={}
for brand in last_ten:
    brand_only= autos[autos["brand"]== brand]
    mean_price= brand_only["price"].mean()
    last_ten_brand[brand]= int(mean_price)
    
last_ten_brand

{'jeep': 11650,
 'subaru': 3953,
 'land_rover': 19108,
 'saab': 3211,
 'jaguar': 11635,
 'daewoo': 1049,
 'trabant': 1790,
 'rover': 1602,
 'lancia': 3376,
 'lada': 2688}

We analyzed the first ten brand and last ten brand. There is interesting price difference. 
<br>
For first ten brand: the price mean is between 2400 and 10000.
<br>
For the last ten brand: the means of prices are either lower than 4000 or higher than 11500.

### Exploring Milage

In [47]:
first_ten_milage={}
for brand in first_ten:
    brand_only= autos[autos["brand"]== brand]
    mean_milage= brand_only["odometer_km"].mean()
    first_ten_milage[brand]=int(mean_milage)
first_ten_milage

{'volkswagen': 128707,
 'bmw': 132572,
 'opel': 129310,
 'mercedes_benz': 130788,
 'audi': 129157,
 'ford': 124266,
 'renault': 128071,
 'peugeot': 127153,
 'fiat': 117121,
 'seat': 121131}

In [48]:
bmm_series= pd.Series(first_ten_milage).sort_values(ascending=False)
print(bmm_series)# Shows the milage of brands

bmw              132572
mercedes_benz    130788
opel             129310
audi             129157
volkswagen       128707
renault          128071
peugeot          127153
ford             124266
seat             121131
fiat             117121
dtype: int64


In [49]:
bpm_series= pd.Series(first_ten_brand).sort_values(ascending= False)
print(bpm_series)#Shows the price of brands

audi             9336
mercedes_benz    8628
bmw              8332
volkswagen       5402
seat             4397
ford             3749
peugeot          3094
opel             2975
fiat             2813
renault          2474
dtype: int64


In [50]:
#Let's create a dataframe from these columns.
First_Ten_Brand=pd.DataFrame(bmm_series, columns=["mean_milage"])
print(First_Ten_Brand)

               mean_milage
bmw                 132572
mercedes_benz       130788
opel                129310
audi                129157
volkswagen          128707
renault             128071
peugeot             127153
ford                124266
seat                121131
fiat                117121


In [51]:
First_Ten_Brand["mean_price"]=bpm_series

First_Ten_Brand

mean_milage  mean_price
bmw                 132572        8332
mercedes_benz       130788        8628
opel                129310        2975
audi                129157        9336
volkswagen          128707        5402
renault             128071        2474
peugeot             127153        3094
ford                124266        3749
seat                121131        4397
fiat                117121        2813

When we look at the amount of the milages, the first think that this brands must be cheapest. However, the situation is different than expected. Still, the price of popular brand is more expensive than the other brands. 